lib/cloudware/azure.rb

```diff
@@ -142,14 +143,30 @@ def machines
                                provider: 'azure',
                                type: r.tags['cloudware_machine_type'],
                                flavour: r.tags['cloudware_machine_flavour'],
-                               state: get_instance_state(r.tags['cloudware_domain'], r.tags['cloudware_machine_name'])
+                               state: machine_state(r.tags['cloudware_machine_name'], r.tags['cloudware_domain'])
                              })
           end
         end
         @machines
       end
     end
 
+    def machine_state(name, domain)
+      state = @compute_client.virtual_machines.instance_view("#{domain}-#{name}", name)
+      return 'running' unless state.statuses.find { |s| s.code =~ /PowerState\/running/ }.nil?
+      return 'stopped' unless state.statuses.find { |s| s.code =~ /PowerState\/stopped/ }.nil?
+    end
+
+    def machine_power_on(name, domain)
+      log.info("#{self.class} Attempting to power off #{name} in #{domain}")
+      @compute_client.virtual_machines.start("#{domain}-#{name}", name)
+    end
+
+    def machine_power_off(name, domain)
+      log.info("#{self.class} Attempting to power on #{name} in #{domain}")
```

Comment:
The log messages here are very similar with the `Aws` class - it might make sense to not duplicate these in both places?